messaging/src/test/java/org/axonframework/messaging/ResourceKeyTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import org.axonframework.messaging.Context.ResourceKey;
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class validating the {@link ResourceKey}.
+ *
+ * @author Steven van Beelen
+ */
+class ResourceKeyTest {
+
+    @Test
+    void resourceKeysShowDebugStringInOutput() {
```

Comment:
Maybe test that two keys with the same label are not considered equal?

Reply:
Smart one. Added a test for this.